roxie/ccd/ccdfile.cpp

```diff
@@ -604,23 +604,33 @@ typedef StringArray *StringArrayPtr;
 
 struct CacheInfoEntry
 {
+    //PageType matches the order of the NodeX enumeration.  Ensure disk entries sort last so that index nodes take precedence.
+    enum PageType
+    {
+        PageTypeBranch = 0,
+        PageTypeLeaf = 1,
+        PageTypeBlob = 2,
+        PageTypeDisk = 3,
+    };
+
     union
     {
         struct
         {
-            bool diskCache : 1;   // false means it's in the jhtree cache, true means it's only in OS disk cache
-            __uint64 page: 39;
-            unsigned file: 24;
+            unsigned type: 2;    // disk or the kind of index node
```

Comment:
Should this be PageType type: 2;

Reply:
I didn't think that would work, but it does.  I will change (and remove the cast elsewhere)